source/Octopus.Tentacle/Kubernetes/ClusterVersion.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Text.RegularExpressions;
+using k8s.Models;
+
+namespace Octopus.Tentacle.Kubernetes
+{
+    public class ClusterVersion : IComparable<ClusterVersion>
+    {
+        public ClusterVersion(int major, int minor)
+        {
+            Major = major;
+            Minor = minor;
+        }
+
+        public int Major { get; }
+        public int Minor { get; }
+
+        public static ClusterVersion FromVersion(Version version)
+        {
+            return new ClusterVersion(version.Major, version.Minor);
+        }
+
+        public static ClusterVersion FromVersionInfo(VersionInfo versionInfo)
+        {
+            return new ClusterVersion(SanitizeAndParseVersionNumber(versionInfo.Major), SanitizeAndParseVersionNumber(versionInfo.Minor));
+        }
+
+        static int SanitizeAndParseVersionNumber(string version)
+        {
+            return int.Parse(Regex.Replace(version, "[^0-9]", ""));
+        }
+
+        public int CompareTo(ClusterVersion? other)
+        {
+            if (other == null) return 1;
+            if (Major > other.Major || (Major == other.Major && Minor > other.Minor)) return 1;
+            if (Major == other.Major && Minor == other.Minor) return 0;
+            return -1;
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (obj is not ClusterVersion clusterVersion)
+                return false;
+
+            return clusterVersion.Major == Major && clusterVersion.Minor == Minor;
+        }
+
+        public override int GetHashCode()
+        {
+#if NET8_0_OR_GREATER
+            return HashCode.Combine(Major, Minor);
+#else
```

Comment:
I'm not sure if there's a way to avoid doing this. I ran into a compiler error similar to the one described [here](https://stackoverflow.com/questions/65941418/cannot-use-new-hashcode-to-override-gethashcode-in-class) because HashCode is not available in the .NET Framework. Since we only build the Kubernetes Tentacle with .NET 8, this part of the code would never actually be used.